### PR TITLE
Merge from desktop and fix Pair selection and token stats.

### PR DESCRIPTION
**Pull request by jsolman**
## Changes
* Merge from Desktop
* Fix Pair selection and token stats

### CODE REVIEW

File: src/components/Dex.vue [hunk 1]
```diff
@@ -98,6 +100,15 @@ export default {
...
     });
 
+
+    this.loadTrades();
+    storeUnwatch = this.$store.watch(
+      () => {
+        return this.$store.state.currentMarket;
+      }, () => {
+        this.loadTrades();
+      });
```
Comment (crishellco) [inline src/components/Dex.vue]: The correct way of doing this is:

```
<script>
import { mapGetters } from 'vuex';

export default {
  ...

  computed: {
    ...mapGetters(['currentMarket']),
  },
  
  ...

  watch: {
    currentMarket() {
      this.loadTrades();
    },
  }
};
</script>
```

File: src/components/dex/Charts.vue [hunk 2]
```diff
@@ -282,20 +287,30 @@ export default {
...
-              onDataCallback(bars, { noData: false })
-            }
+            const resolutionInMinutes = getResolutionInMinutes(resolution);
+
+            this.$store.dispatch('fetchTradesBucketed', {
+              marketName: this.$store.state.currentMarket.marketName,
+              interval: resolutionInMinutes,
+              from: from, 
+              to: to,
+              }).then(() => {
```
Comment (crishellco) [inline src/components/dex/Charts.vue:297]: You can do this here:
```
this.$store.dispatch('fetchTradesBucketed', {
  marketName: this.$store.state.currentMarket.marketName,
  interval: resolutionInMinutes,
  from, 
  to,
}).then(() => {
```

File: src/components/dex/Charts.vue [hunk 3]
```diff
@@ -305,41 +320,43 @@ export default {
...
-              const from = to - 120
+              const to = Math.round(new Date().valueOf() / 1000);
+              const from = to - (resolutionInMinutes * 60 * 2); //  back 2x resolutions
 
-              this.tradingView._options.datafeed.getBars(_symbolInfo, resolution, from, to, (bars) => {
-                if (bars.length === 0) {
-                  return;
-                }
+              const bars = this.$store.state.tradeHistory && this.$store.state.tradeHistory.getBars ?
+                  this.$store.state.tradeHistory.getBars(this.$store.state.tradeHistory, resolution, from, to) :
```
Comment (crishellco) [inline src/components/dex/Charts.vue:327]: Are we storing methods in the Vuex store? That's a whole lot of code smell to me.

File: src/components/dex/Pair.vue [hunk 4]
```diff
@@ -58,19 +58,43 @@ export default {
...
-        const hasTradeHistory = tradeHistory && tradeHistory.trades && tradeHistory.trades.length > 0;
-        const price = {
-          price: this.$formatTokenAmount(hasTradeHistory ? tradeHistory.close24Hour : 0),
-          priceConverted: this.$formatMoney((hasTradeHistory ? tradeHistory.close24Hour : 0) * this.baseCurrencyUnitPrice),
-        };
-        return { asset: quoteCurrency, price, volume: this.quoteVolume, '24H change': this.percentChangeAbsolute };
+        const tickerData = this.tickerDataByMarket[marketName];
+        let price;
+        let quoteVolume;
+        let percentChange;
```
Comment (crishellco) [inline src/components/dex/Pair.vue:66]: You can one-line this.
```
let price, quoteVolume, percentChange;
```

File: src/services/dex.js [hunk 5]
```diff
@@ -1142,11 +1179,24 @@ export default {
     });
   },
 
-  fetchTradesBucketed(marketName, binSize = 1) {
+  fetchTradesBucketed(marketName, binSize = 1, from = null, to = null) {
```
Comment (crishellco) [inline src/services/dex.js:1182]: No need for these default parameter values as they will resolve to `undefined` which is `falsey`.

File: src/services/formatting.js [hunk 6]
```diff
@@ -109,8 +109,13 @@ export default {
     if (nullOrUndefined(value)) {
       return defaultValue;
     }
+    const val = toBigNumber(value);
 
-    return accounting.formatMoney(toBigNumber(value), symbol || settings.getCurrencySymbol());
+    let precision = 2;
+    if (val.isLessThan(1) && !val.isZero()) {
+      precision = 4;
+    }
```
Comment (crishellco) [inline src/services/formatting.js]: Ternary?

File: src/store/mutations.js [hunk 7]
```diff
@@ -160,7 +157,7 @@ function orderBookSnapshotReceived(state, res) {
   const orderBook = dex.formOrderBook(res.asks, res.bids);
   orderBook.pair = res.pair;
 
-  state.orderBook = orderBook;
+  Vue.set(state, 'orderBook', orderBook);
```
Comment (crishellco) [inline src/store/mutations.js:160]: Curious why this change?

File: src/store/mutations.js [hunk 8]
```diff
@@ -335,9 +331,10 @@ function setRecentTransactions(state, transactions) {
   const existingIsEmpty = !state.recentTransactions || state.recentTransactions.length === 0;
 
   _.sortBy(transactions, 'block_index').forEach((transaction) => {
-    const existingTransaction = _.find(state.recentTransactions, ({ hash, symbol }) => {
-      return transaction.hash === hash && transaction.symbol === symbol;
-    });
+    const existingTransaction = _.find(
+      state.recentTransactions,
+      { hash: transaction.hash, symbol: transaction.symbol },
```
Comment (crishellco) [inline src/store/mutations.js:336]: Nice.

**Review by crishellco — APPROVED** [review]
A few minor things, but looks good. The only important thing is the `storeUnwatch` stuff.

File: src/store/mutations.js (hunk 7 above)
Comment (jsolman) [reply src/store/mutations.js:160]: It was in desktop this way. Does this allow Vue to pick up that it changed better?

File: src/services/formatting.js (hunk 6 above)
Comment (jsolman) [reply src/services/formatting.js]: changed

File: src/services/dex.js (hunk 5 above)
Comment (jsolman) [reply src/services/dex.js:1182]: merged this from another's change from desktop. I'll remove the default params.

File: src/components/dex/Pair.vue (hunk 4 above)
Comment (jsolman) [reply src/components/dex/Pair.vue:66]: Actually Lint says "expected variable declaration to be on a new line"; so linter wants it the way it is.

File: src/components/dex/Charts.vue (hunk 3 above)
Comment (jsolman) [reply src/components/dex/Charts.vue:327]: I think it sets a reference to the method. And this is the only place where it happens currently; the reason it needs to happen is related to how the trading chart works. If you deep dive and see a better way to do it, feel free to issue a PR to change it.

File: src/components/dex/Charts.vue (hunk 2 above)
Comment (jsolman) [reply src/components/dex/Charts.vue:297]: adjusted.

File: src/components/Dex.vue (hunk 1 above)
Comment (jsolman) [reply src/components/Dex.vue]: Fixed -- had copied this from desktop -- should fix it there also.
